### PR TITLE
@types/react react css overflow

### PR DESCRIPTION
**Pull request by ddwwcruz**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by ddwwcruz** [comment]
Travis build errors are from material-ui. Possibly in other packages as well.

Should I make a PR that fixes the build errors?

**Comment by sandersn** [comment]
You should fix them in this PR. They happened because of the improved type you added for overflow. I guess material-ui tests depend on react.

**Comment by ddwwcruz** [comment]
Travis is happy now :tada:

File: types/material-ui/material-ui-tests.tsx [hunk 1]
```diff
@@ -109,7 +109,13 @@ function handleTouchTap() {
...
 }
 
-const styles = {
+type Stylesheet<T> = {
+  [P in keyof T]: React.CSSProperties
+};
+
+declare function createStylesheet<T>(args: T): Stylesheet<T>;
+
+const styles = createStylesheet({
```
Comment (vsiao) [inline types/material-ui/material-ui-tests.tsx]: is all of this really necessary? I imagine this should be fixable by just giving `styles` an explicit type annotation, something like

```ts
const style: { [key: string]: React.CSSProperties } = {
    ...
};
```

File: types/react/index.d.ts [hunk 2]
```diff
@@ -1514,12 +1514,12 @@ declare namespace React {
...
          * Controls how extra content exceeding the x-axis of the bounding box of an element is rendered.
          */
-        overflowX?: CSSWideKeyword | any;
+        overflowX?: CSSWideKeyword | 'auto' | 'hidden' | 'scroll' | 'visible';
 
         /**
          * Controls how extra content exceeding the y-axis of the bounding box of an element is rendered.
          */
-        overflowY?: CSSWideKeyword | any;
+        overflowY?: CSSWideKeyword | 'auto' | 'hidden' | 'scroll' | 'visible';
```
Comment (vsiao) [inline types/react/index.d.ts]: prefer double quotes `"`, thanks

File: types/material-ui/material-ui-tests.tsx (hunk 1 above)
Comment (ddwwcruz) [inline types/material-ui/material-ui-tests.tsx]: wasn't aware of that. will work on it.

**Comment by sandersn** [comment]
Looks like Travis has been upgraded today to find even more lint :(

**Review by ericanderson — APPROVED** [review]
I agree with your assessment. This looks ready.

**Comment by sandersn** [comment]
I re-ran Travis and the lint errors went away, so I think they were fixed separately.

**Pull request merged**